unison-cli/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -2899,24 +2865,107 @@ loadPropagateDiffDefaultPatch inputDescription maybeDest0 dest = do
         (ppe, diff) <- diffHelper original (Branch.head patched)
         Cli.respondNumbered (ShowDiffAfterMergePropagate dest0 dest patchPath ppe diff)
 
+delete ::
+  Input ->
+  DeleteOutput ->
+  ((Path.Absolute, HQ'.HQSegment) -> Cli (Set Referent)) -> -- compute matching terms
+  ((Path.Absolute, HQ'.HQSegment) -> Cli (Set Reference)) -> -- compute matching types
+  [Path.HQSplit'] -> -- targets for deletion
+  Cli ()
+delete input doutput getTerms getTypes hqs' = do
+  -- persists the original hash qualified entity for error reporting
+  typesTermsTuple <- traverse (\hq -> do
+      absolute <- Cli.resolveSplit' hq
+      types <- getTypes absolute
+      terms <- getTerms absolute
+      return (hq, types, terms)
+    ) hqs'
+  let notFounds = List.filter (\(_, types, terms) -> Set.null terms && Set.null types) typesTermsTuple
+  -- if there are any entities which cannot be deleted because they don't exist, short circuit.
+  if not $ null notFounds then do
+    let toName :: [(Path.HQSplit', Set Reference, Set referent)] -> [Name]
+        toName notFounds =
+          mapMaybe (\(split,_,_) -> Path.toName' $ HashQualified.toName (HQSplit'.unsplitHQ' split)) notFounds
+    Cli.returnEarly $ NamesNotFound (toName notFounds)
```

Comment:
It looks like it might actually be possible for this list to be empty; what happens in that scenario?

Or is it not possible because it would require the user to have specified just a Hash as an argument?

Reply:
We do a null check for the list length above, but on the off chance the user has inputed something that the function to mapMaybe returns None for, the user would just see no printed names for "did you misspell something?"